UI/Headless/Test.cpp

```diff
@@ -233,6 +234,14 @@ void run_dump_test(HeadlessWebView& view, Test& test, URL::URL const& url, int t
         };
     }
 
+    view.on_set_test_timeout = [&test, timer, timeout_in_milliseconds](double milliseconds) {
+        if (milliseconds == timeout_in_milliseconds)
+            return;
+        timer->stop();
+        auto milliseconds_elapsed = (UnixDateTime::now() - test.start_time).to_milliseconds();
+        timer->start(milliseconds - milliseconds_elapsed);
```

Comment:
I think this crashes whenever the time that has elapsed is larger than the new timeout value (see `EventLoopManagerUnix::register_timer()` for example). Do we really need to account for that instead of resetting the timeout?

Reply:
Good point... I didn't think much about input validation.

I think I was only ever seeing an elapsed time of well under 100ms when testing it. So, just setting a timer of `milliseconds` directly should be fine - I would be very surprised if any test cared about the timeout being precise.